pkg/ottl/ottlfuncs/func_parse_key_value.go

```diff
@@ -0,0 +1,121 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ottlfuncs
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+)
+
+type ParseKeyValueArguments[K any] struct {
+	Target        ottl.StringGetter[K]
+	Delimiter     ottl.Optional[string]
+	PairDelimiter ottl.Optional[string]
+}
+
+func NewParseKeyValueFactory[K any]() ottl.Factory[K] {
+	return ottl.NewFactory("ParseKeyValue", &ParseKeyValueArguments[K]{}, createParseKeyValueFunction[K])
+}
+
+func createParseKeyValueFunction[K any](_ ottl.FunctionContext, oArgs ottl.Arguments) (ottl.ExprFunc[K], error) {
+	args, ok := oArgs.(*ParseKeyValueArguments[K])
+
+	if !ok {
+		return nil, fmt.Errorf("ParseKeyValueFactory args must be of type *ParseKeyValueArguments[K]")
+	}
+
+	return parseKeyValue[K](args.Target, args.Delimiter, args.PairDelimiter)
+}
+
+func parseKeyValue[K any](target ottl.StringGetter[K], d ottl.Optional[string], p ottl.Optional[string]) (ottl.ExprFunc[K], error) {
+	delimiter := "="
+	if !d.IsEmpty() {
+		delimiter = d.Get()
+	}
+
+	pairDelimiter := " "
+	if !p.IsEmpty() {
+		pairDelimiter = p.Get()
+	}
+
+	if pairDelimiter == delimiter {
+		return nil, fmt.Errorf("pair delimiter \"%s\" cannot be equal to delimiter \"%s\"", pairDelimiter, delimiter)
+	}
+
+	return func(ctx context.Context, tCtx K) (any, error) {
+		source, err := target.Get(ctx, tCtx)
+		if err != nil {
+			return nil, err
+		}
+
+		if source == "" {
+			return nil, fmt.Errorf("cannot parse from empty target")
+		}
+
+		pairs, err := splitPairs(source, pairDelimiter)
+		if err != nil {
+			return nil, fmt.Errorf("splitting pairs failed: %w", err)
+		}
+
+		parsed := make(map[string]any)
+		for _, p := range pairs {
+			pair := strings.SplitN(p, delimiter, 2)
+			if len(pair) != 2 {
+				return nil, fmt.Errorf("cannot split '%s' into 2 items, got %d", p, len(pair))
```

Comment:
```suggestion
				return nil, fmt.Errorf("cannot split %q into 2 items, got %d items", p, len(pair))
```
Same here for the format specifier, and just a minor word nit.